File: ios/Classes/FlutterLocalTimezonePlugin.m
#import "FlutterLocalTimezonePlugin.h"
#import <Flutter/Flutter.h>

@implementation FlutterLocalTimezonePlugin

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"flutter_local_timezone"
            binaryMessenger:[registrar messenger]];
  FlutterLocalTimezonePlugin* instance = [[FlutterLocalTimezonePlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getLocalTimezone" isEqualToString:call.method]) {
    NSString* timeZone = [[NSTimeZone localTimeZone] name];
    result(timeZone);
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: ios/Classes/FlutterLocalTimezonePlugin.swift
import Flutter
import UIKit

public class FlutterLocalTimezonePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "flutter_local_timezone", binaryMessenger: registrar.messenger())
    let instance = FlutterLocalTimezonePlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    if call.method == "getLocalTimezone" {
      let timeZone = TimeZone.current.identifier
      result(timeZone)
    } else {
      result(FlutterMethodNotImplemented)
    }
  }
}
